src/condor_schedd.V6/qmgmt.cpp

```diff
@@ -352,6 +352,46 @@ bool operator()(const prio_rec& a, const prio_rec& b) const
 }
 };
 
+// Return the same comparison as above, but just comparing the submitter
+// field.  This allows us to binary search the PrioRecArray by submitter
+// Note the comparison up to the submitter must match the above
+struct prio_rec_submitter_lb {
+bool operator()(const prio_rec& a, const std::string &user) const {
+	if (a.submitter.length() < user.length()) {
+		return false;
+	}
+
+	if (a.submitter.length() > user.length()) {
+		return true;
+	}
+
+	if (a.submitter < user) {
+		return false;
```

Comment:
```suggestion
	if (a.submitter > user) {
		return true;
```
Good point, Tim. I missed that. The code needs this change.